### PR TITLE
I just hopes all this shit works fine

### PR DESCRIPTION
**Pull request by Mrjoulin**

### CODE REVIEW

File: src/main/kotlin/su/arlet/business1/security/sevices/AuthUsersDetails.kt [hunk 1]
```diff
@@ -0,0 +1,59 @@
+package su.arlet.business1.security.sevices
```
Comment (Arlet2) [inline src/main/kotlin/su/arlet/business1/security/sevices/AuthUsersDetails.kt]: sevices -> services

File: src/main/kotlin/su/arlet/business1/security/sevices/AuthUsersDetailsService.kt [hunk 2]
```diff
@@ -0,0 +1,25 @@
...
+
+import su.arlet.business1.exceptions.UserNotFoundException
+import su.arlet.business1.repos.UserRepo
+import su.arlet.business1.security.jwt.JwtUtils
+
+@Service
+class AuthUsersDetailsService(
+    private val userRepository: UserRepo,
+) : UserDetailsService {
+    @Transactional
```
Comment (Arlet2) [inline src/main/kotlin/su/arlet/business1/security/sevices/AuthUsersDetailsService.kt]: ???

File: src/main/kotlin/su/arlet/business1/security/jwt/JwtUtils.kt [hunk 3]
```diff
@@ -0,0 +1,71 @@
...
+import su.arlet.business1.exceptions.UnauthorizedError
+import su.arlet.business1.security.sevices.AuthUsersDetails
+import java.time.Instant
+import java.util.*
+import javax.crypto.SecretKey
+
+@Component
+class JwtUtils {
+    @Value("\${app.jwtSecret}")
+    private var jwtSecret: String = ""
```
Comment (Arlet2) [inline src/main/kotlin/su/arlet/business1/security/jwt/JwtUtils.kt:19]: set something simple

Comment (Arlet2) [inline src/main/kotlin/su/arlet/business1/security/jwt/JwtUtils.kt:18]: maybe add default value?

File: src/main/kotlin/su/arlet/business1/security/jwt/AuthTokenFilter.kt [hunk 4]
```diff
@@ -0,0 +1,69 @@
...
+            val userDetails = getUserDetails(request)
+            if (userDetails != null) {
+                val authentication = UsernamePasswordAuthenticationToken(
+                    userDetails, null, userDetails.authorities
+                )
+                authentication.details = WebAuthenticationDetailsSource().buildDetails(request)
+                SecurityContextHolder.getContext().authentication = authentication
+            }
+        } catch (e: Exception) {
+            Companion.logger.error("Cannot set user authentication: { }", e)
```
Comment (Arlet2) [inline src/main/kotlin/su/arlet/business1/security/jwt/AuthTokenFilter.kt]: what is { } ? I hope it's okey for this logger

File: src/main/kotlin/su/arlet/business1/security/jwt/AuthEntryPoint.kt [hunk 5]
```diff
@@ -0,0 +1,40 @@
...
+    ) {
+
+        logger.error("Unauthorized error: {}", authException.message)
+
+        response.contentType = MediaType.APPLICATION_JSON_VALUE
+        response.status = HttpServletResponse.SC_UNAUTHORIZED
+
+        val body: MutableMap<String, Any?> = HashMap()
+
+        body["status"] = HttpServletResponse.SC_UNAUTHORIZED
```
Comment (Arlet2) [inline src/main/kotlin/su/arlet/business1/security/jwt/AuthEntryPoint.kt]: Need to change to our error messages type

File: src/main/kotlin/su/arlet/business1/security/SecurityConfig.kt [hunk 6]
```diff
@@ -0,0 +1,114 @@
...
+    @Bean
+    fun passwordEncoder(): PasswordEncoder {
+        return BCryptPasswordEncoder()
+    }
+
+    @Bean
+    fun roleHierarchy(): RoleHierarchy {
+        val roleHierarchy = RoleHierarchyImpl()
+        roleHierarchy.setHierarchy(
+            "ROLE_ADMIN > ROLE_EDITOR\n" +
```
Comment (Arlet2) [inline src/main/kotlin/su/arlet/business1/security/SecurityConfig.kt:72]: Can change \n to """ or something like that. I forget

File: src/main/kotlin/su/arlet/business1/security/jwt/AuthTokenFilter.kt [hunk 7]
```diff
@@ -0,0 +1,69 @@
...
+        try {
+            val userDetails = getUserDetails(request)
+            if (userDetails != null) {
+                val authentication = UsernamePasswordAuthenticationToken(
+                    userDetails, null, userDetails.authorities
+                )
+                authentication.details = WebAuthenticationDetailsSource().buildDetails(request)
+                SecurityContextHolder.getContext().authentication = authentication
+            }
+        } catch (e: Exception) {
```
Comment (Arlet2) [inline src/main/kotlin/su/arlet/business1/security/jwt/AuthTokenFilter.kt:39]: Need to handle user not found exception cause it's a filter and controller advice cannot help us for it

File: src/main/kotlin/su/arlet/business1/controllers/ArticleController.kt [hunk 8]
```diff
@@ -102,6 +114,7 @@ class ArticleController(
     }
 
     @PatchMapping("/{id}/status")
+    @PreAuthorize("hasRole('EDITOR')")
```
Comment (Arlet2) [inline src/main/kotlin/su/arlet/business1/controllers/ArticleController.kt:116]: How I can changed status from NEED_FIXES -> APPROVED if I'm a journalist?

File: src/main/kotlin/su/arlet/business1/security/sevices/AuthUsersDetails.kt (hunk 1 above)
Comment (Mrjoulin) [reply src/main/kotlin/su/arlet/business1/security/sevices/AuthUsersDetails.kt]: OK

File: src/main/kotlin/su/arlet/business1/security/SecurityConfig.kt (hunk 6 above)
Comment (Mrjoulin) [reply src/main/kotlin/su/arlet/business1/security/SecurityConfig.kt:72]: Not needed

File: src/main/kotlin/su/arlet/business1/security/jwt/AuthTokenFilter.kt (hunk 4 above)
Comment (Mrjoulin) [reply src/main/kotlin/su/arlet/business1/security/jwt/AuthTokenFilter.kt]: Fixed

File: src/main/kotlin/su/arlet/business1/controllers/ArticleController.kt (hunk 8 above)
Comment (Mrjoulin) [reply src/main/kotlin/su/arlet/business1/controllers/ArticleController.kt:116]: Когда журналист делает патч с обновлением текста или названия автоматически ставится в ON_REVIEW

**Pull request merged**